src/main/java/org/zalando/riptide/RestBase.java

```diff
@@ -77,4 +88,43 @@ public D execute(final HttpMethod method, final URI url, final HttpHeaders heade
 
     protected abstract <T> D execute(final HttpMethod method, final URI url, final HttpEntity<T> entity);
 
+    static <T> void writeRequestEntity(final HttpEntity<T> entity, final ClientHttpRequest request, List<HttpMessageConverter<?>> converters) throws IOException {
```

Comment:
One thing that `RestTemplate` currently does is setting accept headers based on the registered message converters. That logic would have to be reimplemented here.